omx/gstomxaudioenc.c

```diff
@@ -822,21 +867,46 @@ gst_omx_audio_enc_set_format (GstAudioEncoder * encoder, GstAudioInfo * info)
       return FALSE;
   } else {
     if (!(klass->cdata.hacks & GST_OMX_HACK_NO_DISABLE_OUTPORT)) {
-      /* Disable output port */
-      if (gst_omx_port_set_enabled (self->enc_out_port, FALSE) != OMX_ErrorNone)
+      GST_DEBUG_OBJECT (self, "Enabling ports");
+
+      if (gst_omx_port_set_enabled (self->enc_in_port, TRUE) != OMX_ErrorNone)
+        return FALSE;
+
+      if (gst_omx_port_wait_enabled (self->enc_in_port,
+              1 * GST_SECOND) != OMX_ErrorNone)
+        return FALSE;
+
+      /* Enable output port */
```

Comment:
Try to be a little bit more consistent with comments, for example previous gst_omx_port_set_enabled call does not have a 'Enable input port' comment